File: internal/content/testdata/bench/java/nc-longest-substring.java
import java.util.*;
class Solution {
    public long length_of_longest_substring(String s) {
        Map<Character, Integer> lastSeen = new HashMap<>();
        int left = 0, best = 0;
        for (int right = 0; right < s.length(); right++) {
            char ch = s.charAt(right);
            if (lastSeen.containsKey(ch) && lastSeen.get(ch) >= left)
                left = lastSeen.get(ch) + 1;
            lastSeen.put(ch, right);
            if (right - left + 1 > best) best = right - left + 1;
        }
        return best;
    }
}

File: internal/content/testdata/bench/typescript/nc-longest-substring.ts
function length_of_longest_substring(s: string): number {
    const lastSeen: {[ch: string]: number} = {};
    let left = 0;
    let best = 0;
    for (let right = 0; right < s.length; right++) {
        const ch = s[right];
        if (ch in lastSeen && lastSeen[ch] >= left) {
            left = lastSeen[ch] + 1;
        }
        lastSeen[ch] = right;
        if (right - left + 1 > best) {
            best = right - left + 1;
        }
    }
    return best;
}
